compiler/src/dotty/tools/dotc/typer/TypeAssigner.scala

```diff
@@ -398,6 +398,9 @@ trait TypeAssigner {
         .appliedTo(defn.QuotesClass.typeRef, defn.QuotedExprClass.typeRef.appliedTo(tpt.tpe))
     tree.withType(lambdaType)
 
+  def assignType(tree: untpd.SplicedExpr, tpt: Tree)(using Context): SplicedExpr =
+    tree.withType(tpt.tpe)
```

Comment:
If tpt is always the same as the type of the node, and if tpt is always EmptyTree in untyped trees, then it shouldn't be necessary and can be removed (possibly using `promote` in TreeChecker to retype it).

Reply:
I was wondering how the `tpt` could be removed. I will try that.

Reply:
We currently use the `tpt` in the `staging`/`splicing` phases to set staged healed types for quotes and splices. I did not manage to remove it without breaking something in those phases.

Reply:
> I did not manage to remove it without breaking something in those phases.

We have other phases where we transform types of trees like https://github.com/lampepfl/dotty/blob/main/compiler/src/dotty/tools/dotc/transform/ElimErasedValueType.scala, does that not work here? Maybe I can have a look if you show me what the issues you ran into were.

Reply:
I managed to remove the `tpt`.